src/keyboard.c

```diff
@@ -40,7 +40,9 @@ static void
 end_cycling(struct server *server)
 {
 	desktop_focus_and_activate_view(&server->seat, server->osd_state.cycle_view);
-	desktop_move_to_front(server->osd_state.cycle_view);
+	if (server->osd_state.cycle_view) {
+		view_move_to_front(server->osd_state.cycle_view);
```

Comment:
This could use the _no_focus variant as `osd_finish()` also calls `cursor_update_focus()`.

Reply:
`_no_focus` variant removed now, so marking this resolved.